File: mobile/app/src/main/java/com/archi/subscription/data/model/Payment.kt
package com.archi.subscription.data.model

data class Payment(
    val id: String,
    val subscriptionId: String,
    val serviceProvider: String = "",
    val subscriberNumber: String = "",
    val amount: Double,
    val paymentDate: String,
    val period: String,
    val status: String,
    val transactionReference: String? = null
)

data class CreatePaymentRequest(
    val subscriptionId: String,
    val amount: Double,
    val period: String
)

File: backend/src/ArchiSubscription.Core/Entities/Payment.cs
using ArchiSubscription.Core.Enums;

namespace ArchiSubscription.Core.Entities;

/// <summary>
/// Represents a payment made for a specific subscription in a specific billing period.
/// </summary>
public class Payment : BaseEntity
{
    public Guid SubscriptionId { get; set; }
    public decimal Amount { get; set; }
    public DateTime PaymentDate { get; set; }
    public string Period { get; set; } = string.Empty; // Format: "2026-05"
    public PaymentStatus Status { get; set; }
    public string? TransactionReference { get; set; }

    // Navigation property
    public Subscription Subscription { get; set; } = null!;
}

